browser/views.py

```diff
@@ -220,31 +221,55 @@ def _handle_known_person_form(request, extra_form, person):
 
 
 @staff_member_required
-def edit_person(request, person_id):
-    person = get_object_or_404(Person, pk=person_id)
+def handle_person(request, person_id=None):
+    template = "browser/change_person.html"
+    context = {}
+    person = None
+
+    if person_id:
+        person = get_object_or_404(Person, pk=person_id)
+
     if request.method == 'GET':
-        form = PersonForm(instance=person)
-        extra_form = KnownPersonForm(instance=person.authority, prefix='known')
+        if person_id:
+            form = PersonForm(instance=person)
+            extra_form = KnownPersonForm(instance=person.authority, prefix='known')
+        else:
+            form = PersonForm(initial={'changed_by': request.user})
```

Comment:
you should set the user after the form is submitted; otherwise it could be changed before before the form is sent back to the server

Reply:
For setting the user correctly I added a single line as shown below.
```
if form.is_valid():
    form.cleaned_data["changed_by"] = request.user
    person = form.save()
```
Before the form.save, changed_by will be explicitly set to request.user.